compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -364,6 +390,10 @@ impl SelfProfilerRef {
         }
     }
 
+    pub fn get_or_alloc_cached_string(&self, s: &'static str) -> Option<StringId> {
```

Comment:
Can you add a doc string to this method? Something like
```
/// Gets a `StringId` for the given string. This method makes sure that
/// any strings going through it will only be allocated once in the
/// profiling data. Returns `None` if the self-profiling is not enabled.
```

Reply:
Also, I think this can be just `&str` instead of `&'static str`.